DBSCE/Overview.md

```diff
@@ -0,0 +1,219 @@
+
+### Device Registration Client
+
+This is a pre-requisite for DBSC(E) to work.
+
+Device Registration Client is a process where the user registers the device with the IdP and is expected to be a once-in-a-lifetime operation.
+
+The device registration establishes trust between the device and a service that maintains a directory of all devices. This document does not cover the protocol of device registration, but it assumes that during device registration, some asymmetric keys are shared between the client and the service, typically a device key and some other keys necessary for the secure device communication. A client software component that performs the device registration is called a _device registration client_. As mentioned above, the key assumption in DBSC(E) is that device registration happened in a clean room environment, and it is the responsibility of the device owner to ensure this.
+
+One device registration client can manage multiple devices on the same physical device. There also can be multiple device registration clients on the same device. The device registration client can be owned and supported by:
+
+- Operating system - the device gets registered when the OS is installed.
+- Browser - the device gets registered when the browser is installed.
+- Device management software (MDM provider) - the device gets registered when the MDM is enrolled.
+- Third-party software vendor - the device gets registered according to the vendor rules.
+
+DBSC(E) aims to support most of these scenarios. It does not define the device registration protocol amd is only concerned with the keys generated in a "clean room" state and the management of the generated keys to prove device binding.
+
+### Local Key Helper
+
+DBSC(E) introduced the concept of `Local Key Helper` which can be mapped to the `TPM` or any `Key generation helper` for the consumer case.
+**Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. It can be Public or Private and is expected to be either shipped as a part of a given enterprise framework (with the IdP/OS) or can be installed by a provider in compliance with the protocol expanded below.
+
+From the deployment point of view there are two types of local key helpers: _well-known_(_private_) and _third party_(_public_)
+
+- _Public local key helper_ or _third party_: Can be accessed by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP through a well-defined public protocol. A third party local key helper has a special deployment mechanism.
```

Comment:
"through a well-defined public protocol" do we have one in mind to suggest here or does it not yet exist?

Reply:
We are adding details in the `platform` section, added windows, and mac in review. May be I can link to the section for better clarity.

Reply:
Also, may be we can rephrase this. Two parts here: 
1. API(IDP calls Local Key Helper to generateKeys with HelperId) is defined here in this document and will be formalized in the specification.
2. Specifics of Local Key Helper are added in the `Platform examples` below.

Rephrased the text above, please check.